packages/app/app/actions/downloads.js

```diff
@@ -1,53 +1,119 @@
 import { safeAddUuid } from './helpers';
+import { store } from '../persistence/store';
 
+export const READ_DOWNLOADS = 'READ_DOWNLOADS';
 export const ADD_TO_DOWNLOADS = 'ADD_TO_DOWNLOADS';
 export const DOWNLOAD_STARTED = 'DOWNLOAD_STARTED';
 export const DOWNLOAD_PROGRESS = 'DOWNLOAD_PROGRESS';
 export const DOWNLOAD_FINISHED = 'DOWNLOAD_FINISHED';
 export const DOWNLOAD_ERROR = 'DOWNLOAD_ERROR';
 export const CLEAR_FINISHED_DOWNLOADS = 'CLEAR_FINISHED_DOWNLOADS';
 
-export function addToDownloads(streamProviders, track) {  
+const changePropertyForItem = ({downloads, uuid, propertyName='status', value}) => {
```

Comment:
> redux middleware that subscribes to changes in the downloads subtree

do you mean a generic redux middleware which filters download action to set downloads in store or..?

Reply:
Yes, something like that.
For a different project, I wrote a middleware for localstorage persistence (reusing code from another library), which looked like this:
https://github.com/nukeop/tinycrawl-frontend/blob/master/src/redux/localstorage.js

We could do something like that, but save the current state of the download queue in the electron-store file instead of localstorage.

Reply:
done. That was a nice piece of work btw.